src/vcpkg/commands.extract.cpp

```diff
@@ -0,0 +1,191 @@
+#include <vcpkg/base/fwd/message_sinks.h>
+
+#include <vcpkg/base/files.h>
+#include <vcpkg/base/strings.h>
+
+#include <vcpkg/archives.h>
+#include <vcpkg/commands.extract.h>
+#include <vcpkg/commands.h>
+#include <vcpkg/vcpkgcmdarguments.h>
+#include <vcpkg/vcpkgpaths.h>
+
+namespace vcpkg::Commands
+{
+    static constexpr StringLiteral OPTION_STRIP = "strip";
+    static constexpr StringLiteral OPTION_EXTRACT_TYPE = "extract-type";
+
+    static constexpr std::array<CommandSetting, 2> EXTRACT_SETTINGS = {{
+        {OPTION_STRIP, []() { return msg::format(msgStripOption, msg::option = "strip"); }},
+        {OPTION_EXTRACT_TYPE, []() { return msg::format(msgExtractTypeOption, msg::option = "extract-type"); }},
+    }};
+
+    const CommandStructure ExtractCommandStructure = {
+        [] { return msg::format(msgExtractHelp); },
+        2,
+        3,
+        {{}, {EXTRACT_SETTINGS}, {}},
+        nullptr,
+    };
+
+    static ExtractionType get_extraction_type(const ParsedArguments& options)
+    {
+        static const std::unordered_map<std::string, ExtractionType> extraction_type_map = {
+            {"tar", ExtractionType::TAR},
+            {"zip", ExtractionType::ZIP},
+            {"nupkg", ExtractionType::NUPKG},
+            {"msi", ExtractionType::MSI},
+            {"exe", ExtractionType::EXE},
```

Comment:
The purpose of those settings it to "override" detection that's based on the file extension. I.e., if you wanted to unpack a vsix file you could just say `--extract-type=zip` and it will just try to unzip it regardless of the extension.

Reply:
I removed the `--extract-type` option. It now just falls back to cmake if it encounters a unique extension.